### PR TITLE
Introduce Optuna Dashboard in tutorial docs

### PR DESCRIPTION
**Pull request by c-bata**
## Motivation

To increase the number of visitors to the Optuna Dashboard.

## Description of the changes

Introduce Optuna Dashboard in the tutorial docs.

Screenshot 2022-12-08 18 10 18

https://optuna--4226.org.readthedocs.build/en/4226/tutorial/10_key_features/005_visualization.html

### CODE REVIEW

**Comment by c-bata** [comment]
@keisuke-umezawa Could you review this PR?

**Comment by cross32768** [comment]
The overall description is LGTM, but I found one problem. And I also have one small comment.

Below is the problem.

```
def objective(trial):
    data, target = sklearn.datasets.load_breast_cancer(return_X_y=True)
    train_x, valid_x, train_y, valid_y = train_test_split(data, target, test_size=0.25)
    dtrain = lgb.Dataset(train_x, label=train_y)
    dvalid = lgb.Dataset(valid_x, label=valid_y)

    param = {
        "objective": "binary",
        "metric": "auc",
        "verbosity": -1,
        "boosting_type": "gbdt",
        "bagging_fraction": trial.suggest_float("bagging_fraction", 0.4, 1.0),
        "bagging_freq": trial.suggest_int("bagging_freq", 1, 7),
        "min_child_samples": trial.suggest_int("min_child_samples", 5, 100),
    }

    # Add a callback for pruning.
    pruning_callback = optuna.integration.LightGBMPruningCallback(trial, "auc")
    gbm = lgb.train(
        param, dtrain, valid_sets=[dvalid], verbose_eval=False, callbacks=[pruning_callback]
    )

    preds = gbm.predict(valid_x)
    pred_labels = np.rint(preds)
    accuracy = sklearn.metrics.accuracy_score(valid_y, pred_labels)
    return accuracy
```

```
study = optuna.create_study(
    direction="maximize",
    sampler=[optuna.samplers.TPESampler](https://docs.python.org/3/library/abc.html#abc.ABC)(seed=[SEED](https://docs.python.org/3/library/functions.html#int)),
    pruner=[optuna.pruners.MedianPruner](https://docs.python.org/3/library/abc.html#abc.ABC)(n_warmup_steps=10),
)
study.optimize(objective, n_trials=100, timeout=600)
```

In the current tutorial's preview, this code outputs below. Maybe it doesn't expected behavior.
```
'verbose_eval' argument is deprecated and will be removed in a future release of LightGBM. Pass 'log_evaluation()' callback via 'callbacks' argument instead.
'verbose_eval' argument is deprecated and will be removed in a future release of LightGBM. Pass 'log_evaluation()' callback via 'callbacks' argument instead.
'verbose_eval' argument is deprecated and will be removed in a future release of LightGBM. Pass 'log_evaluation()' callback via 'callbacks' argument instead.
'verbose_eval' argument is deprecated and will be removed in a future release of LightGBM. Pass 'log_evaluation()' callback via 'callbacks' argument instead.
'verbose_eval' argument is deprecated and will be removed in a future release of LightGBM. Pass 'log_evaluation()' callback via 'callbacks' argument instead.
'verbose_eval' argument is deprecated and will be removed in a future release of LightGBM. Pass 'log_evaluation()' callback via 'callbacks' argument instead.
'verbose_eval' argument is deprecated and will be removed in a future release of LightGBM. Pass 'log_evaluation()' callback via 'callbacks' argument instead.
'verbose_eval' argument is deprecated and will be removed in a future release of LightGBM. Pass 'log_evaluation()' callback via 'callbacks' argument instead.
'verbose_eval' argument is deprecated and will be removed in a future release of LightGBM. Pass 'log_evaluation()' callback via 'callbacks' argument instead.
'verbose_eval' argument is deprecated and will be removed in a future release of LightGBM. Pass 'log_evaluation()' callback via 'callbacks' argument instead.
'verbose_eval' argument is deprecated and will be removed in a future release of LightGBM. Pass 'log_evaluation()' callback via 'callbacks' argument instead.
'verbose_eval' argument is deprecated and will be removed in a future release of LightGBM. Pass 'log_evaluation()' callback via 'callbacks' argument instead.
'verbose_eval' argument is deprecated and will be removed in a future release of LightGBM. Pass 'log_evaluation()' callback via 'callbacks' argument instead.
'verbose_eval' argument is deprecated and will be removed in a future release of LightGBM. Pass 'log_evaluation()' callback via 'callbacks' argument instead.
```

Below is the small comment.

* How about specifying storage for study? It looks that some change is needed to use sample code with `optuna-dashboard`.

**Review by cross32768 — COMMENTED** [review]
The overall description is LGTM, but I found one problem. And I also have one small comment.

Below is the problem.

```
def objective(trial):
    data, target = sklearn.datasets.load_breast_cancer(return_X_y=True)
    train_x, valid_x, train_y, valid_y = train_test_split(data, target, test_size=0.25)
    dtrain = lgb.Dataset(train_x, label=train_y)
    dvalid = lgb.Dataset(valid_x, label=valid_y)

    param = {
        "objective": "binary",
        "metric": "auc",
        "verbosity": -1,
        "boosting_type": "gbdt",
        "bagging_fraction": trial.suggest_float("bagging_fraction", 0.4, 1.0),
        "bagging_freq": trial.suggest_int("bagging_freq", 1, 7),
        "min_child_samples": trial.suggest_int("min_child_samples", 5, 100),
    }

    # Add a callback for pruning.
    pruning_callback = optuna.integration.LightGBMPruningCallback(trial, "auc")
    gbm = lgb.train(
        param, dtrain, valid_sets=[dvalid], verbose_eval=False, callbacks=[pruning_callback]
    )

    preds = gbm.predict(valid_x)
    pred_labels = np.rint(preds)
    accuracy = sklearn.metrics.accuracy_score(valid_y, pred_labels)
    return accuracy
```

```
study = optuna.create_study(
    direction="maximize",
    sampler=[optuna.samplers.TPESampler](https://docs.python.org/3/library/abc.html#abc.ABC)(seed=[SEED](https://docs.python.org/3/library/functions.html#int)),
    pruner=[optuna.pruners.MedianPruner](https://docs.python.org/3/library/abc.html#abc.ABC)(n_warmup_steps=10),
)
study.optimize(objective, n_trials=100, timeout=600)
```

In the current tutorial's preview, this code outputs below. Maybe it doesn't expected behavior.
```
'verbose_eval' argument is deprecated and will be removed in a future release of LightGBM. Pass 'log_evaluation()' callback via 'callbacks' argument instead.
'verbose_eval' argument is deprecated and will be removed in a future release of LightGBM. Pass 'log_evaluation()' callback via 'callbacks' argument instead.
'verbose_eval' argument is deprecated and will be removed in a future release of LightGBM. Pass 'log_evaluation()' callback via 'callbacks' argument instead.
'verbose_eval' argument is deprecated and will be removed in a future release of LightGBM. Pass 'log_evaluation()' callback via 'callbacks' argument instead.
'verbose_eval' argument is deprecated and will be removed in a future release of LightGBM. Pass 'log_evaluation()' callback via 'callbacks' argument instead.
'verbose_eval' argument is deprecated and will be removed in a future release of LightGBM. Pass 'log_evaluation()' callback via 'callbacks' argument instead.
'verbose_eval' argument is deprecated and will be removed in a future release of LightGBM. Pass 'log_evaluation()' callback via 'callbacks' argument instead.
'verbose_eval' argument is deprecated and will be removed in a future release of LightGBM. Pass 'log_evaluation()' callback via 'callbacks' argument instead.
'verbose_eval' argument is deprecated and will be removed in a future release of LightGBM. Pass 'log_evaluation()' callback via 'callbacks' argument instead.
'verbose_eval' argument is deprecated and will be removed in a future release of LightGBM. Pass 'log_evaluation()' callback via 'callbacks' argument instead.
'verbose_eval' argument is deprecated and will be removed in a future release of LightGBM. Pass 'log_evaluation()' callback via 'callbacks' argument instead.
'verbose_eval' argument is deprecated and will be removed in a future release of LightGBM. Pass 'log_evaluation()' callback via 'callbacks' argument instead.
'verbose_eval' argument is deprecated and will be removed in a future release of LightGBM. Pass 'log_evaluation()' callback via 'callbacks' argument instead.
'verbose_eval' argument is deprecated and will be removed in a future release of LightGBM. Pass 'log_evaluation()' callback via 'callbacks' argument instead.
```

Below is the small comment.

* How about specifying `storage` for `study`? It looks that some change is needed to use sample code with `optuna-dashboard`.

**Comment by c-bata** [comment]
@cross32768 Thank you for your review!

> In the current tutorial's preview, this code outputs below. Maybe it doesn't expected behavior.

Do you think the deprecated messages should be addressed in this PR? or can we work on it in a separated PR?

> How about specifying storage for study? It looks that some change is needed to use sample code with optuna-dashboard.

I am concerned about using SQLite3 in this sample code since this is executed when building the documentation. It creates `db.sqlite3` on the file system and make the documentation build fragile on re-run. What do you think?

**Comment by cross32768** [comment]
I think it is good to work with other PR.
> Do you think the deprecated messages should be addressed in this PR? or can we work on it in a separated PR?

I see. If there is such a situation, I think that the code is better as it is. On the other hand, I think to add a guidance about specifying storage to `study` to use tutorial's code with`optuna-dashboard` may be helpful for practitioner. What do you think?
> I am concerned about using SQLite3 in this sample code since this is executed when building the documentation. It creates db.sqlite3 on the file system and make the documentation build fragile on re-run. What do you think?

**Comment by c-bata** [comment]
> I see. If there is such a situation, I think that the code is better as it is. On the other hand, I think to add a guidance to explain that we need specify storage to use tutorial's code withoptuna-dashboard may be helpful for practitioner. What do you think?

That sounds good. However, this note is not the main topic of the visualization tutorial, so let me just remove the commands and guide users to check the README of optuna-dashboard.

be4f0f91f

PTAL.

File: tutorial/10_key_features/005_visualization.py [hunk 1]
```diff
@@ -10,6 +10,26 @@
 
 For visualizing multi-objective optimization (i.e., the usage of :func:`optuna.visualization.plot_pareto_front`),
 please refer to the tutorial of :ref:`multi_objective`.
+
+.. note::
+   By using `Optuna Dashboard <https://github.com/optuna/optuna-dashboard>`_, you can also check the optimization history,
+   hyperparameter importances, hyperparameter relationships, etc. in graphs and tables.
+   Please persistent your study using :ref:`RDB backend <rdb>` and execute following commands to run Optuna Dashboard.
```
Comment (cross32768) [inline tutorial/10_key_features/005_visualization.py:17]: nits.
```suggestion
   Please make your study persistent using :ref:`RDB backend <rdb>` and execute following commands to run Optuna Dashboard.
```

**Review by cross32768 — APPROVED** [review]
LGTM! Thank you for improvement of description (and I apologize for making a slightly off-topic comment).

I have a nitpick about grammar, but I approve this PR at this time for smooth merge.

**Comment by c-bata** [comment]
@toshihikoyanase Could you review this PR?

**Review by toshihikoyanase — APPROVED** [review]
I confirmed that the note was rendered as expected. LGTM.

[Note] I'm not sure of the right place of the note. Another choice is the bottom of this page, but I guess few people will notice the note since this tutorial is relatively long with many graphs. So, I'd like to accept the current change.

**Pull request merged**